pkg/endpointslice/map.go

```diff
@@ -32,16 +34,16 @@ type endpointInfo struct {
 }
 
 type clusterInfo struct {
-	hostIPs map[string][]string
-	ipList  []string
+	hostRecords map[string][]serviceimport.DNSRecord
+	recordList  []serviceimport.DNSRecord
 }
 
 type Map struct {
 	epMap map[string]*endpointInfo
 	sync.RWMutex
 }
 
-func (m *Map) GetIPs(hostname, cluster, namespace, name string, checkCluster func(string) bool) ([]string, bool) {
+func (m *Map) GetIPs(hostname, cluster, namespace, name string, checkCluster func(string) bool) ([]serviceimport.DNSRecord, bool) {
```

Comment:
```suggestion
func (m *Map) GetDNSRecords(hostname, cluster, namespace, name string, checkCluster func(string) bool) ([]serviceimport.DNSRecord, bool) {
```
sorry for the later nit: can we change the name ? Not really important we can do it later.